File: cursethedarkness.py
import math
m = int(input())
for i in range(m):
    bx, by = map(float, input().split())
    c = int(input())
    ok = False
    for j in range(c):
        cx, cy = map(float, input().split())
        if math.sqrt((bx-cx)*(bx-cx) + (by-cy)*(by-cy)) <= 8:
            ok = True
    if ok:
        print("light a candle")
    else:
        print("curse the darkness")

File: cursethedarkness.cpp
#include <stdio.h>
#include <string.h>
#include <cmath>
using namespace std;

int main() {
	int m, c;
	float bx, by, cx, cy;
	scanf("%d", &m);
	while (m--) {
		scanf("%f %f %d", &bx, &by, &c);
		bool ok = false;
		while (c--) {
			scanf("%f %f", &cx, &cy);
			if (sqrt((bx - cx) * (bx - cx) + (by - cy) * (by - cy)) <= 8)
				ok = true;
		}

		if (ok)
			printf("light a candle\n");
		else
			printf("curse the darkness\n");
	}
	return 0;
}